worksheet.rb

```diff
@@ -31,4 +134,92 @@
 # - the total amount of money each driver has made
 # - the average rating for each driver
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# - Which driver has the highest average rating?
+puts "The number of rides each driver has given:"
+
+def number_of_rides(all_driver_info)
+  all_driver_info.each do |driver, ride|
+    puts "#{driver}: #{ride.length}"
+  end
+end
+
+number_of_rides(all_driver_info)
```

Comment:
Here, you wrap your loops in a method each time. however, I don't think this gets you much in terms of reducing code! look for spots where code is repeated to make the best use of methods.